x-pack/plugins/enterprise_search/public/applications/app_search/components/role_mappings/role_mappings_logic.ts

```diff
@@ -0,0 +1,337 @@
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { clearFlashMessages, flashAPIErrors } from '../../../shared/flash_messages';
+import { HttpLogic } from '../../../shared/http';
+import { KibanaLogic } from '../../../shared/kibana';
+import { ANY_AUTH_PROVIDER } from '../../../shared/role_mapping/constants';
+import { AttributeName } from '../../../shared/types';
+import { ROLE_MAPPINGS_PATH } from '../../routes';
+import { ASRoleMapping, RoleTypes } from '../../types';
+import { roleHasScopedEngines } from '../../utils/role/has_scoped_engines';
+import { Engine } from '../engine/types';
+
+import { DELETE_ROLE_MAPPINGS_MESSAGE } from './constants';
+
+interface RoleMappingsServerDetails {
+  roleMappings: ASRoleMapping[];
+  multipleAuthProvidersConfig: boolean;
+}
+
+interface RoleMappingServerDetails {
+  attributes: string[];
+  authProviders: string[];
+  availableEngines: Engine[];
+  elasticsearchRoles: string[];
+  hasAdvancedRoles: boolean;
+  multipleAuthProvidersConfig: boolean;
+  roleMapping?: ASRoleMapping;
+}
+
+const getFirstAttributeName = (roleMapping: ASRoleMapping) =>
+  Object.entries(roleMapping.rules)[0][0] as AttributeName;
+const getFirstAttributeValue = (roleMapping: ASRoleMapping) =>
+  Object.entries(roleMapping.rules)[0][1] as AttributeName;
+
+export interface RoleMappingsActions {
+  handleAccessAllEnginesChange(): void;
+  handleAuthProviderChange(value: string[]): { value: string[] };
+  handleAttributeSelectorChange(
+    value: AttributeName,
+    firstElasticsearchRole: string
+  ): { value: AttributeName; firstElasticsearchRole: string };
+  handleAttributeValueChange(value: string): { value: string };
+  handleDeleteMapping(): void;
+  handleEngineSelectionChange(
+    engineName: string,
+    selected: boolean
+  ): {
+    engineName: string;
+    selected: boolean;
+  };
+  handleResetMappings(callback: () => void): Function;
+  handleRoleChange(roleType: RoleTypes): { roleType: RoleTypes };
+  handleSaveMapping(): void;
+  initializeRoleMapping(roleId?: string): { roleId?: string };
+  initializeRoleMappings(): void;
+  resetState(): void;
+  setRoleMappingData(data: RoleMappingServerDetails): RoleMappingServerDetails;
+  setRoleMappingsData(data: RoleMappingsServerDetails): RoleMappingsServerDetails;
+}
+
+export interface RoleMappingsValues {
+  accessAllEngines: boolean;
+  attributeName: AttributeName;
+  attributeValue: string;
+  attributes: string[];
+  availableAuthProviders: string[];
+  availableEngines: Engine[];
+  dataLoading: boolean;
+  elasticsearchRoles: string[];
+  hasAdvancedRoles: boolean;
+  multipleAuthProvidersConfig: boolean;
+  roleMapping: ASRoleMapping | null;
+  roleMappings: ASRoleMapping[];
+  roleType: RoleTypes;
+  selectedAuthProviders: string[];
+  selectedEngines: Set<string>;
+}
+
+export const RoleMappingsLogic = kea<MakeLogicType<RoleMappingsValues, RoleMappingsActions>>({
```

Comment:
[NOT a change request since this is just an as-is migration, just a general comment for the future] This logic file is a decent amount to parse/wrap one's head around. It's also a bit of a flag to me when the tests for a file is so large that GitHub auto-collapses the diff 😅 

Just my 2c, I'd like to see us split logic files up more modularly where it makes sense (since Kea does such a great job of making it easy to connect logic files nowadays). For this particular file for example, if I were writing it from scratch I would probably split it up into `RoleMappingsLogic` and `RoleMappingLogic` (singular).

Definitely interested to hear your thoughts on that and if you like the concept!